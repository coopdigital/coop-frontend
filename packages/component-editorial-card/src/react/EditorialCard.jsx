import React from "react";
import PropTypes from "prop-types";
// import '../css/EditorialCard.pcss';

const EditorialCard = ({ title, text, link, type, testId }) => {
  const tagAttributes = {
    className: "coop coop-c-editorialcard coop-c-editorialcard--horizontal",
    role: type === "warn" || type === "error" ? "alert" : "status",
  };

  return (
    <div
      data-testid={testId}
      className={tagAttributes.className}
      role={tagAttributes.role}
    >
      <div className="coop-c-editorialcard__inner ">
        <h3 className="coop-c-editorialcard__title coop-c-notification__heading">{title}</h3>
        <p className="coop-c-notification__p">
          {link ? (
            <a href={link} className="coop-c-notification__link">
              {text}
            </a>
          ) : (
            text
          )}
        </p>
      </div>
    </div>
  );
};

EditorialCard.defaultProps = {
  type: "info",
};

EditorialCard.propTypes = {
  title: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  link: PropTypes.string,
  type: PropTypes.oneOf(["info", "warn", "error", "success"]),
  testId: PropTypes.string,
};

export default EditorialCard;
