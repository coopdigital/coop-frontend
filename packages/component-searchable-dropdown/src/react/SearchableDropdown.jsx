import React, { useRef, useEffect } from "react";
import SearchableDropdownLibrary from "../lib/SearchableDropdownLibrary.js";

const SearchableDropdown = () => {
  const comboboxRef = useRef();

  useEffect(() => {
    new SearchableDropdownLibrary(comboboxRef.current).setup();
  }, []);

  return (
    <div ref={comboboxRef}>
      <div data-combobox>
        <label htmlFor="combobox">How you know them</label>
        <input
          name="combobox"
          className="coop-form__input"
          type="text"
          placeholder="Select option"
          data-input
          data-testid="combobox-input"
          role="combobox"
          aria-controls="comboboxvalues"
          aria-autocomplete="list"
          aria-expanded="false"
        />
        <div
          style={{ display: "none" }}
          data-dropdown
          data-testid="combobox-dropdown"
        >
          <ul id="comboboxvalues" role="listbox" data-list>
            {/* These valuse would be mapped from a data prop */}
            <li role="option" aria-selected="false" data-option>
              Partner
            </li>
            <li role="option" aria-selected="false" data-option>
              Spouse
            </li>
            <li role="option" aria-selected="false" data-option>
              Parent
            </li>
            <li role="option" aria-selected="false" data-option>
              Child
            </li>
            <li role="option" aria-selected="false" data-option>
              Brother
            </li>
            <li role="option" aria-selected="false" data-option>
              Grandchild
            </li>
            <li role="option" aria-selected="false" data-option>
              Father
            </li>
          </ul>
        </div>
      </div>
    </div>
  );
};

export default SearchableDropdown;
