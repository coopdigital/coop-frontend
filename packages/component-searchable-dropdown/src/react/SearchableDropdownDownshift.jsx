/* eslint-disable indent */
import React, { useState } from 'react';
import { useCombobox } from 'downshift';
import PropTypes from 'prop-types';
import { matchSorter } from 'match-sorter';

const SearchableDropdown = ({
  className,
  compact,
  id,
  label,
  noResults,
  onSelect,
  options,
  placeholder,
  style,
}) => {
  const [inputItems, setInputItems] = useState(options);
  const sizeVariant = compact ? 'compact' : '';
  const noResultsText = noResults || 'No relevant options';
  const placeholderValue = placeholder || null;

  const {
    isOpen,
    closeMenu,
    getToggleButtonProps,
    getLabelProps,
    getMenuProps,
    getInputProps,
    getComboboxProps,
    openMenu,
    getItemProps,
  } = useCombobox({
    id,
    items: inputItems,
    onInputValueChange: ({ inputValue }) => {
      setInputItems(matchSorter(options, inputValue));
    },
    onSelectedItemChange: ({ selectedItem: selected }) => {
      if (typeof onSelect === 'function') onSelect(selected);
      closeMenu();
    },
  });

  return (
    <div className={`coop-c-combobox ${sizeVariant} ${className}`} style={style}>
      <label {...getLabelProps()} htmlFor={`${id}-input`}>
        {label}
      </label>
      <div {...getComboboxProps()}>
        <input {...getInputProps({ onFocus: openMenu })} placeholder={placeholderValue} />
        <button type="button" {...getToggleButtonProps()} aria-label="toggle menu">
          <svg xmlns="http://www.w3.org/2000/svg" height="24" width="24">
            <path d="m12 15.375-6-6 1.4-1.4 4.6 4.6 4.6-4.6 1.4 1.4Z" />
          </svg>
        </button>
      </div>
      <ul {...getMenuProps()}>
        {isOpen && inputItems.length
          ? inputItems.map((item, index) => (
              <li {...getItemProps({ item, index })} key={`${item}${index}`}>
                {item}
              </li>
            ))
          : null}
        {isOpen && !inputItems.length ? (
          <li className="coop-c-combobox--null">{noResultsText}</li>
        ) : null}
      </ul>
    </div>
  );
};

export { SearchableDropdown };

SearchableDropdown.propTypes = {
  className: PropTypes.string,
  compact: PropTypes.bool,
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  noResults: PropTypes.string,
  onSelect: PropTypes.func,
  options: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.number, PropTypes.string])).isRequired,
  placeholder: PropTypes.string,
  style: PropTypes.object,
};
