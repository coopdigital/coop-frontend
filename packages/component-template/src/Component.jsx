import React from 'react';
import PropTypes from 'prop-types';
import './component.pcss';

// Example react component with 3 props, using prop-types for type management.
const Component = ({ className, heading, content }) => (
  <article>
    <h3 className={className}>{heading}</h3>
    <div dangerouslySetInnerHTML={{ __html: content }} />
  </article>
);

Component.defaultProps = {
  className: null,
  heading: null,
  content: null,
};

Component.propTypes = {
  className: PropTypes.string,
  heading: PropTypes.string,
  content: PropTypes.node,
};

export default Component;
