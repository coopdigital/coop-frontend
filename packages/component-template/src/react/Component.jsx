import React from "react";

const Component = () => <div class="coop-c-component">Example HTML</div>;

export default Component;
