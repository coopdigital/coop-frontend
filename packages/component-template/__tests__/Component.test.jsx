/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Component from '../src/index.mjs';

afterEach(cleanup);

describe('Component', () => {
  it('should render default Component', () => {
    const wrapper = render(<Component heading="heading" />);
    expect(() => wrapper.unmount()).not.toThrow();

    const { asFragment } = render(<Component heading="heading" />);
    expect(asFragment(<Component heading="heading" />)).toMatchSnapshot();
  });
});
