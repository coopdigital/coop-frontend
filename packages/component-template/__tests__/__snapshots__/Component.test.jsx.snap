// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should render default Component 1`] = `
<DocumentFragment>
  <article>
    <h3>
      heading
    </h3>
    <div />
  </article>
</DocumentFragment>
`;
